src/components/materialization/GenerateButton.tsx

```diff
@@ -136,19 +136,63 @@ function MaterializeGenerateButton({ disabled, callFailed }: Props) {
                 { overrideJsonFormDefaults: true }
             );
 
-            const newDraftId = draftsResponse.data[0].id;
+            let evaluatedDraftId = persistedDraftId;
+            let taskDraftExists = false;
+
+            if (persistedDraftId) {
+                const existingDraftSpecResponse =
+                    await getDraftSpecsByCatalogName(
+                        persistedDraftId,
+                        entityName,
+                        'materialization'
+                    );
+
+                if (existingDraftSpecResponse.error) {
+                    return callFailed({
+                        error: {
+                            title: 'materializationCreate.generate.failure.errorTitle',
+                            error: existingDraftSpecResponse.error,
+                        },
+                    });
+                } else if (
+                    existingDraftSpecResponse.data &&
+                    existingDraftSpecResponse.data.length > 0
+                ) {
+                    taskDraftExists = true;
+                }
+            } else {
+                const draftsResponse = await createEntityDraft(entityName);
+
+                if (draftsResponse.error) {
+                    return callFailed({
+                        error: {
+                            title: 'materializationCreate.generate.failure.errorTitle',
+                            error: draftsResponse.error,
+                        },
+                    });
+                }
+
+                evaluatedDraftId = draftsResponse.data[0].id;
+            }
+
             const draftSpec = generateDraftSpec(
                 encryptedEndpointConfig.data,
                 imagePath,
                 resourceConfig
             );
 
-            const draftSpecsResponse = await createDraftSpec(
-                newDraftId,
-                entityName,
-                draftSpec,
-                'materialization'
-            );
+            const draftSpecsResponse =
+                persistedDraftId && taskDraftExists
+                    ? await modifyDraftSpec(draftSpec, {
+                          draft_id: evaluatedDraftId,
```

Comment:
With the addition of the task draft existence flag, I am _tempted_ to say it is safe to add the catalog name match condition back to this update draft spec call.